ex2_system/source/diagnostic/diagnosic.c

```diff
@@ -54,22 +54,24 @@ static void uhf_watchdog_daemon(void *pvParameters) {
         if (err != U_GOOD_CONFIG) {
             ex2_log("UHF was not responsive - attempting to toggle power.");
             // Toggle the UHF.
-            const unsigned int timeout = 30 * ONE_SECOND; // 30 seconds
+            const unsigned int timeout = 5 * ONE_SECOND; // 5 seconds
+            unsigned int attempts = 0;
             eps_set_pwr_chnl(UHF_PWR_CHNL, OFF);
             TickType_t start = xTaskGetTickCount();
 
-            while (eps_get_pwr_chnl(UHF_PWR_CHNL) != OFF && xTaskGetTickCount() - start < timeout) {
+            while ((eps_get_pwr_chnl(UHF_PWR_CHNL) != OFF) || ((xTaskGetTickCount() - start) < timeout)) {
                 vTaskDelay(ONE_SECOND);
-            }
-
-            if (eps_get_pwr_chnl(UHF_PWR_CHNL) != OFF) {
-                ex2_log("UHF failed to power off.");
+                if(attempts >= 10){
+                    ex2_log("UHF failed to power off.");
+                    break;
+                }
+                attempts++;

```

Comment:
This is redundant with the timeout mechanism.

Reply:
That's fair. It should be removed.

Reply:
Done